velox/functions/lib/ArrayShuffle.cpp

```diff
@@ -112,12 +117,41 @@ std::vector<std::shared_ptr<exec::FunctionSignature>> signatures() {
               .build()};
 }
 
-} // namespace
+std::vector<std::shared_ptr<exec::FunctionSignature>> arrayShuffleWithCustomSeedSignatures() {
+    return {exec::FunctionSignatureBuilder()
+              .typeVariable("T")
+              .returnType("array(T)")
+              .argumentType("array(T)")
+              .constantArgumentType("bigint")
+              .build()};
+}
+
+exec::VectorFunctionMetadata arrayShuffleMetadata() {
+  return exec::VectorFunctionMetadataBuilder()
+      .deterministic(false)
```

Comment:
It seems like Spark's version is deterministic, no?

Reply:
Updated, thanks.

Reply:
Sorry my bad, I think it should be marked as non-deterministic to prevent constant folding. If it's deterministic, the output for `wrapInConstant(3, [1, 2, 3])` would be `{[3, 2, 1], [3, 2, 1], [3, 2, 1]}`, this is not expected. Please let me know if this makes sense to you.

Reply:
@zhli1142015 I see. Makes sense.